File: AbetOrder/AbetOrder/AbetOrder.Entity/OrderPdf.cs
using OpenDataAccess.Entity;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace AbetOrder.Entity
{
    [TableProperty(Name = "t_d_OrderPdf", PrimaryKey = "Id")]
    public class OrderPdf : EntityModel, IEntity
    {
        /// <summary> 
        /// 主键
        /// </summary> 
        public Guid Id { get; set; }

        public Guid CreateUser { get; set; }

        public Guid OrderId { get; set; }
        /// <summary> 
        /// 路径
        /// </summary> 
        public string PdfPath { get; set; }
        /// <summary> 
        /// 失败信息
        /// </summary> 
        public string FailMessage { get; set; }
        /// 1：成功 2：失败
        /// </summary> 
        public byte GenStatus { get; set; }
        /// <summary> 
        /// 创建时间
        /// </summary> 
        public DateTime CreateDate { get; set; }

        ///  1:订单，2：加工单
        /// </summary> 
        public byte PdfType { get; set; }
    }

    [TableProperty(Name = "v_OrderPdf", PrimaryKey = "Id")]
    public class ViewOrderPdf : OrderPdf
    {
        public int FailId { get; set; }

        public string CreateUserName { get; set; }

        public byte DataRight { get; set; }

        public string GenStatusName { get; set; }

        public string PdfTypeName { get; set; }

        public string OrderName2 { get; set; }
    }
}

File: AbetOrder/AbetOrder/AbetOrder.WebConfig/configs/OrderPdf.js
(function () {
    window.configs.OrderPdf = {
        Name: "OrderPdf",
        Title: "订单PDF",
        EntityName: "OrderPdf",
        PrimaryKey: "Id",
        TemplateName: "EntityListPage",
        UpdateStatusUrl: "OrderPdf/ReGenPdf",
        SelectNames: ["Id", "OrderName2", "FailId", "GenStatus", "PdfPath", "FailMessage", "CreateUserName", "GenStatusName", "PdfTypeName", "CreateDate"],
        SearchNames: ["PdfType", "GenStatus", "OrderName2"],
        DataColumnNames: ["OrderName2", "PdfTypeName", "GenStatusName", "CreateUserName", "CreateDate"],
        OrderByList: [{ Name: "CreateDate", IsDesc: true }],
        Properties: GetProperties(),
        DefaultConditions: GetDefaultConditions(),
        IsNewAdd: false,
        IsUpdate: true,
        IsDelete: false,
        TableWidth: 1000
    };

    function GetDefaultConditions() {
        return [{
            Name: "DataRight",
            Type: "int",
            IsCurrentUser: true,
            PropertyName: "DataRight"
        }]
    }

    function GetProperties() {
        return [{
            Label: "类型", Name: "PdfType", AllowClear: true, OperateLogic: "=",
            DataType: "byte", Type: "Select", DataSource: GetPdfTypeDataSource()
        },
        {
            Label: "状态", Name: "GenStatus",
            DataType: "int", Type: "Select", DataSource: GetStatusDataSource(), OperateLogic: "=", AllowClear: true
        },
        { Label: "订单", Name: "OrderName2", IsOpenPage: true, IsRandom: false, SearchProperty: { PropertyName: null }, PropertyName: "PdfPath", PageUrl: "{PdfPath}", DataType: "string", PlaceHolder: "模糊匹配订单编号或门板花式" },
        { Label: "类型", Name: "PdfTypeName" },
        { Label: "创建时间", Name: "CreateDate" },
        { Label: "创建人", Name: "CreateUserName" },
        { Label: "状态", Name: "GenStatusName" }]
    }

    function GetStatusDataSource() {
        return [{ Value: "1", Text: "成功" }, { Value: "2", Text: "失败" }]
    }

    function GetPdfTypeDataSource() {
        return [{ Value: "1", Text: "订单" }, { Value: "2", Text: "加工单" }]
    }

})();

